Src/ILGPU/Backends/OpenCL/CLInstructions.cs

```diff
@@ -54,19 +55,22 @@ public static partial class CLInstructions
         /// Resolves an unary arithmetic operation.
         /// </summary>
         /// <param name="kind">The arithmetic kind.</param>
-        /// <param name="isFloat">True, if this is a floating-point operation.</param>
+        /// <param name="basicValueType">The arithmetic basic value type.</param>
         /// <param name="isFunction">
         /// True, if the resolved operation is a function call.
         /// </param>
         /// <returns>The resolved arithmetic operation.</returns>
         public static string GetArithmeticOperation(
             UnaryArithmeticKind kind,
-            bool isFloat,
+            ArithmeticBasicValueType basicValueType,
             out bool isFunction)
         {
-            if (!UnaryArithmeticOperations.TryGetValue(
-                (kind, isFloat),
-                out var operation))
+            var category = basicValueType.IsFloat()
```

Comment:
Can we create a mapping that immediately maps `ArtihmeticBasicValueType` values to their appropriate `CLUnaryCategory` values? I guess this makes it easier to spot where and how those value instances are being created 🔍